core/src/main/java/org/jruby/RubyBoolean.java

```diff
@@ -163,7 +163,14 @@ public static IRubyObject false_xor(ThreadContext context, IRubyObject fals, IRu
 
         @JRubyMethod(name = "to_s", alias = "inspect")
         public static RubyString false_to_s(ThreadContext context, IRubyObject fals) {
-            return RubyString.newStringShared(context.runtime, FALSE_BYTES);
+            Ruby runtime = context.runtime;
+            RubyString toRubyString = RubyString.newStringShared(runtime, FALSE_BYTES);
+
+            if (runtime.getInstanceConfig().isFrozenCoreToS()) {
+                return runtime.freezeAndDedupString(toRubyString);
```

Comment:
unless its cached we're just doing more work on every `to_s` - this wasn't the intention of the original PR

Reply:
My understanding is that the string is being cached but I could be wrong, of course: https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/Ruby.java#L4587-L4662

What you mean by original PR, #5868 or MRI one?

Reply:
right, the point being that there was a field check instead of a map lookup.